### DISCUSSION TITLE
Discord export schedule via docker container.

### DISCUSSION THREAD

**Discussion by Peterkal2112** [Tyrrrz/DiscordChatExporter#593] [Q&A]
Hello,
I found this tutorial: https://github.com/Tyrrrz/DiscordChatExporter/wiki/Scheduling-exports-with-Cron
But it only deals with chat exporter in windows or Linux and not docker.. I made a docker chat exporter command that is working fine when I run it via SSH, it is working even when I run a script that command is included in it, but it is not working scheduled when i want to run that script via Crontab -e.
Can you help me ?

**Comment by Yudi** [comment]
Hi!
Can you share the `discordchatexporter.log` and `discordchatexportererror.log` logs located in the `/tmp` directory?
You can read the files by running the following commands:

```cat /tmp/discordchatexporter.log```
```cat /tmp/discordchatexportererror.log```

**Reply by Peterkal2112** [reply]
Hello,
Unfortunately I don't have these log files, my DiscordChatExporter running in Docker container and I don't really know how to check these files in docker container which is running only when a command is initiated via ssh.
sincerely.

**Reply by Yudi** [reply]
If I understood it correctly, you're trying to schedule the cron job and run the script on the host machine, not inside the Docker container.

If the above is correct, either (1) you aren't redirecting the output to a log or (2) the cron job isn't running.
<br>
**1.** Check if your `crontab -e` settings match the format below:


```
* * * * * /path/to/cron.sh > /tmp/discordchatexporter.log 2> /tmp/discordchatexportererror.log
```

Where:
`* * * * *` is the time
`/path/to/cron.sh` is the path to your script
`> /tmp/discordchatexporter.log` is the command to redirect the output to the /tmp directory
`2> /tmp/discordchatexportererror.log` is the command to redirect errors to the /tmp directory
<br>
**2.** If you are redirecting the output correctly, but the logs aren't being created, check the CRON logs:

```
grep CRON /var/log/syslog
```

If the cron job is running successfully, it should show up in the log like this:
```
May 22 20:32:01 ubuntu CRON[20554]: (root) CMD (/path/to/cron.sh > /tmp/discordchatexporter.log 2> /tmp/discordchatexportererror.log)
```

**Reply by Peterkal2112** [reply]
Hello,
I tried that your method but no luck.
My script looks like this:
"
#! /bin/sh
docker run --rm -it -v /media/PathToDisk/Discord_Servers/:/app/out tyrrrz/discordchatexporter:stable exportguild -g IDofAserver -t MyUserToken -o "%G - %b" -f HtmlDark --dateformat "dd.MM.yyyy HH:mm:ss.ffff"
"

I tried also that script that is on github page: https://github.com/Tyrrrz/DiscordChatExporter/wiki/Scheduling-exports-with-Cron
But cannot tell to the script where to look for dll path at DLLFOLDER= (because my enviroment is linux and docker container).

when i run my script in terminal it will run without any problem byt dont want to run scheduled as a cronjob.

The output from cat /tmp/discordchatexportererror.log
is
"the input device is not a TTY"

there is no output from cat /tmp/discordchatexporter.log

I do not know what to do at this point...

In the documentation there is no mention about Docker, cronjob schedule from docker container....

sincerely,

**Reply by Yudi** [reply]
Try removing the `-it` flag from your command:

```
docker run --rm -v /media/PathToDisk/Discord_Servers/:/app/out tyrrrz/discordchatexporter:stable exportguild -g IDofAserver -t MyUserToken -o "%G - %b" -f HtmlDark --dateformat "dd.MM.yyyy HH:mm:ss.ffff"
```
